pstoremgr/pstoremgr.go

```diff
@@ -136,29 +133,47 @@ func (pm *Manager) PeersAddresses(peers []peer.ID) []api.Multiaddr {
 		return nil
 	}
 
-	var addrs []api.Multiaddr
+	var pinfos []peerstore.PeerInfo
 	for _, p := range peers {
 		if p == pm.host.ID() {
 			continue
 		}
-		addrs = append(addrs, pm.filteredPeerAddrs(p)...)
+		pinfo := peerstore.PeerInfo{
+			ID:    p,
+			Addrs: pm.filteredPeerAddrs(p),
+		}
+		if len(pinfo.Addrs) > 0 {
+			pinfos = append(pinfos, pinfo)
+		}
 	}
-	return addrs
+
+	toSort := &peerSort{
+		pinfos: pinfos,
+		pstore: pm.host.Peerstore(),
+	}
+	// Sort from highest to lowest priority
+	sort.Sort(toSort)
+
+	return toSort.pinfos
 }
 
 // ImportPeers calls ImportPeer for every address in the given slice, using the
-// given connect parameter.
-func (pm *Manager) ImportPeers(addrs []ma.Multiaddr, connect bool) error {
-	for _, a := range addrs {
-		pm.ImportPeer(a, connect)
+// given connect parameter. Peers are tagged with priority as given
+// by their position in the list.
+func (pm *Manager) ImportPeers(addrs []ma.Multiaddr, connect bool, ttl time.Duration) error {
+	for i, a := range addrs {
+		pid, err := pm.ImportPeer(a, connect, ttl)
+		if err == nil {
+			pm.SetPriority(pid, i)
```

Comment:
The first peer in the slice gets priority 0 (highest). Loading the peerstore uses this so the addresses at the top of the peerstore get the highest priorities and will be saved in the same order. Bootstrap() also uses these priorities to contact peers starting from the highest priority (0).